packages/core/src/Table/stories/TableHooks/UseHvRowState.tsx

```diff
@@ -0,0 +1,93 @@
+import { useMemo } from "react";
+import {
+  HvTable,
+  HvTableBody,
+  HvTableContainer,
+  HvTableHead,
+  HvTableHeader,
+  HvTableRow,
+  HvTableCell,
+  useHvData,
+  useHvRowState,
+  HvInput,
+  HvIconButton,
+} from "@hitachivantara/uikit-react-core";
+
+import { Close, Edit } from "@hitachivantara/uikit-react-icons";
+
+import { makeData, getColumns, AssetEvent } from "../storiesUtils";
+
+const EditableCell = ({ value }) => <HvInput defaultValue={value} />;
+
+export const UseHvRowState = () => {
+  const columns = useMemo(
+    () => [
+      ...getColumns(),
+      {
+        id: "edit",
+        Cell: (props) => {
+          const { row, setRowState } = props;
+          return (
+            <HvIconButton
+              title={row.state.isEditing ? "Close" : "Edit"}
+              variant="secondaryGhost"
+              aria-label="edit button"
+              onClick={() =>
+                setRowState?.([row.id], (state) => ({
+                  ...state,
+                  isEditing: !state.isEditing,
+                }))
+              }
+            >
+              {row.state.isEditing ? <Close /> : <Edit />}
+            </HvIconButton>
+          );
+        },
+      },
+    ],
+    []
+  );
+  const data = useMemo(() => makeData(6), []);
+
+  const { getTableProps, getTableBodyProps, prepareRow, headerGroups, rows } =
+    useHvData<AssetEvent, string>({ columns, data }, useHvRowState);
+
+  return (
+    <HvTableContainer>
+      <HvTable {...getTableProps()}>
+        <HvTableHead>
+          {headerGroups.map((headerGroup) => (
+            <HvTableRow {...headerGroup.getHeaderGroupProps()}>
+              {headerGroup.headers.map((col) => (
+                <HvTableHeader {...col.getHeaderProps()}>
```

Comment:
we're gonna need this
```suggestion
                <HvTableHeader {...col.getHeaderProps()} aria-hidden={col.variant === "actions" ? true : undefined}>
```